### ISSUE TITLE
Do not discard return value of fgets in function ‘ya_str_get_default_net_interface’

### ISSUE THREAD

**Issue by jacksgt** [yabar-dev/yabar#3]
Code:
```c
       if(route_h) {
                //Ignore first line as it is headers
                fgets(buffer, sizeof(buffer), route_h);
                fgets(buffer, sizeof(buffer), route_h);
                char *it = (char*) memchr(buffer, '\t', strlen(buffer));
                *it = '\0';
        }
```
(see https://github.com/yabar-dev/yabar/blob/master/src/intern_blks/ya_intern.c#L220)

Compiler:
```
src/intern_blks/ya_intern.c: In function ‘ya_str_get_default_net_interface’:
src/intern_blks/ya_intern.c:220:22: warning: ignoring return value of ‘fgets’, declared with attribute warn_unused_result [-Wunused-result]
                 fgets(buffer, sizeof(buffer), route_h);
                      ^
src/intern_blks/ya_intern.c:221:22: warning: ignoring return value of ‘fgets’, declared with attribute warn_unused_result [-Wunused-result]
                 fgets(buffer, sizeof(buffer), route_h);
```
see https://travis-ci.org/yabar-dev/yabar/builds/191465528#L197

Even though the first line is discarded, the first fgets may still have failed (basically resulting in undefined behaviour) **and** we don't even check the output of the second (*relevant*) fgets.

http://man7.org/linux/man-pages/man3/fgets.3p.html

**Comment by erebe** [comment]
I made a pass over warnings here https://github.com/yabar-dev/yabar/pull/5
I didn't care much about returns of fgets as we read a file from procfs filesystem which is in RAM. And if fgets fails, well the user will have bigger problems than yabbar.

**Comment by jacksgt** [comment]
Fixed with #5
